projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-filtering.component.ts

```diff
@@ -487,37 +487,34 @@ export class IgxGridExcelStyleFilteringComponent extends BaseFilteringComponent
     }
 
     private renderValues() {
-        this.filterValues = this.generateFilterValues(this.column.dataType === GridColumnDataType.Date || this.column.dataType === GridColumnDataType.DateTime);
+        this.filterValues = this.generateFilterValues();
         this.generateListData();
     }
 
-    private generateFilterValues(isDateColumn = false) {
-        let filterValues;
+    private generateFilterValues() {
+        const formatValue = (value: any): any => {
+            if (!value) return value;
 
-        if (isDateColumn) {
-            filterValues = new Set<any>(this.expressionsList.reduce((arr, e) => {
-                if (e.expression.condition.name === 'in') {
-                    return [...arr, ...Array.from((e.expression.searchVal as Set<any>).values()).map(v =>
-                        new Date(v).toISOString())];
-                }
-                return [...arr, ...[e.expression.searchVal ? e.expression.searchVal.toISOString() : e.expression.searchVal]];
-            }, []));
-        } else if (this.column.dataType === GridColumnDataType.Time) {
-            filterValues = new Set<any>(this.expressionsList.reduce((arr, e) => {
-                if (e.expression.condition.name === 'in') {
-                    return [...arr, ...Array.from((e.expression.searchVal as Set<any>).values()).map(v =>
-                        typeof v === 'string' ? v : new Date(v).toLocaleTimeString())];
-                }
-                return [...arr, ...[e.expression.searchVal ? e.expression.searchVal.toLocaleTimeString() : e.expression.searchVal]];
-            }, []));
-        } else {
-            filterValues = new Set<any>(this.expressionsList.reduce((arr, e) => {
-                if (e.expression.condition.name === 'in') {
-                    return [...arr, ...Array.from((e.expression.searchVal as Set<any>).values())];
-                }
-                return [...arr, ...[e.expression.searchVal]];
-            }, []));
-        }
+            switch (this.column.dataType) {
+                case GridColumnDataType.Date:
+                    return new Date(value).toDateString();
+                case GridColumnDataType.DateTime:
+                    return new Date(value).toISOString();
+                case GridColumnDataType.Time:
+                    return typeof value === 'string' ? value : new Date(value).toLocaleTimeString();
+                default:
+                    return value;
+            }
+        };
+
+        const processExpression = (arr: any[], e: any): any[] => {
+            if (e.expression.condition.name === 'in') {
```

Comment:
Not sure, but maybe we should add the 'not in' condition here as well?

Reply:
When filter is applied and there more than two selected items in the `IgxExcelStyleSearchComponent`, a single filtering operand with 'in' condition and a set of search values is created. So here we are only checking for that 'in' condition to handle the `searchVal` properly.